Mathlib/MeasureTheory/Group/MeasurableEquiv.lean

```diff
@@ -0,0 +1,258 @@
+/-
+Copyright (c) 2021 Yury G. Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury G. Kudryashov
+
+! This file was ported from Lean 3 source module measure_theory.group.measurable_equiv
+! leanprover-community/mathlib commit 95413e23e3d29b45c701fcd31f2dbadaf1b79cba
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.MeasureTheory.Group.Arithmetic
+
+/-!
+# (Scalar) multiplication and (vector) addition as measurable equivalences
+
+In this file we define the following measurable equivalences:
+
+* `MeasurableEquiv.smul`: if a group `G` acts on `α` by measurable maps, then each element `c : G`
+  defines a measurable automorphism of `α`;
+* `MeasurableEquiv.vadd`: additive version of `MeasurableEquiv.smul`;
+* `MeasurableEquiv.smul₀`: if a group with zero `G` acts on `α` by measurable maps, then each
+  nonzero element `c : G` defines a measurable automorphism of `α`;
+* `MeasurableEquiv.mulLeft`: if `G` is a group with measurable multiplication, then left
+  multiplication by `g : G` is a measurable automorphism of `G`;
+* `MeasurableEquiv.addLeft`: additive version of `MeasurableEquiv.mulLeft`;
+* `MeasurableEquiv.mulRight`: if `G` is a group with measurable multiplication, then right
+  multiplication by `g : G` is a measurable automorphism of `G`;
+* `MeasurableEquiv.addRight`: additive version of `MeasurableEquiv.mulRight`;
+* `MeasurableEquiv.mulLeft₀`, `MeasurableEquiv.mulRight₀`: versions of
+  `MeasurableEquiv.mulLeft` and `MeasurableEquiv.mulRight` for groups with zero;
+* `MeasurableEquiv.inv`: `Inv.inv` as a measurable automorphism
+  of a group (or a group with zero);
+* `MeasurableEquiv.neg`: negation as a measurable automorphism of an additive group.
+
+We also deduce that the corresponding maps are measurable embeddings.
+
+## Tags
+
+measurable, equivalence, group action
+-/
+
+namespace MeasurableEquiv
+
+variable {G G₀ α : Type _} [MeasurableSpace G] [MeasurableSpace G₀] [MeasurableSpace α] [Group G]
+  [GroupWithZero G₀] [MulAction G α] [MulAction G₀ α] [MeasurableSMul G α] [MeasurableSMul G₀ α]
+
+/-- If a group `G` acts on `α` by measurable maps, then each element `c : G` defines a measurable
+automorphism of `α`. -/
+@[to_additive (attr := simps! (config := { fullyApplied := false }) toEquiv apply)
+      "If an additive group `G` acts on `α` by measurable maps, then each element `c : G`
+      defines a measurable automorphism of `α`." ]
```

Comment:
I should ask for particular reviewer attention to these lines. I don't understand all of the arguments and I mostly just shuffled things around until error messages stop showing up. I'm not sure if I preserved the intended result.

The same thing happens later with `MeasurableEquiv.inv`

Reply:
Looks good to me.